dataloaders/objectron_main.py

```diff
@@ -0,0 +1,184 @@
+''' Parse objectron data to PyTorch dataloader. Cereal box for now only with shuffled images.'''
+from pathlib import Path
+import os
+from torch.utils.data import Dataset, DataLoader
+import torch
+import cv2 as cv
+import json
+import numpy as np
+from icecream import ic
+from numba import jit
+import albumentations as A
+import sys
+
+sys.path.insert(1, '/home/prokofiev/Objectron')
+from objectron.schema import features
+from objectron.dataset import box, graphics
+
+
+class Objectron(Dataset):
+    def __init__(self, root_folder, mode='train', transform=None, debug_mode=False):
+        self.root_folder = root_folder
+        self.transform = transform
+        self.debug_mode = debug_mode
+        self.mode = mode
+        if mode == 'train':
+            ann_path = Path(root_folder).resolve() /  'annotations/objectron_train.json'
+            with open(ann_path, 'r') as f:
+                self.ann = json.load(f)
+        elif mode in ['val', 'test']:
+            ann_path = Path(root_folder).resolve() /  'annotations/objectron_test.json'
+            with open(ann_path, 'r') as f:
+                self.ann = json.load(f)
+        else:
+            raise RuntimeError("Unknown dataset mode")
+
+    def __len__(self):
+        return len(self.ann['annotations'])
+
+    def __getitem__(self, indx):
+        # get path to image from annotations
+        raw_keypoints = self.ann['annotations'][indx]['keypoints']
+        img_id = self.ann['annotations'][indx]['image_id']
+        category = self.ann['annotations'][indx]['category_id']
+        # get raw key points for bb from annotations
+        img_path = self.root_folder + '/' + (self.ann['images'][img_id]['file_name'])
+        # read image
+        image = cv.imread(img_path, flags=1)
+        assert image is not None
+        # The keypoints are [x, y] where `x` and `y` are unnormalized
+        # transform raw key points to this representation
+        unnormalized_keypoints = np.array(raw_keypoints).reshape(9, 2)
+        # "print" image after crop with keypoints if needed
+        if self.debug_mode:
+            imgh = image.copy()
+            b = np.zeros((9,3))
+            b[:,:2] = self.normalize(image, unnormalized_keypoints)
+            graphics.draw_annotation_on_image(imgh, b , [9])
+            cv.imwrite('image_before_pipeline.jpg', imgh)
+        # given unnormalized keypoints crop object on image
+        cropped_keypoints, cropped_img = self.crop(image, unnormalized_keypoints)
+
+        # convert colors from BGR to RGB
+        cropped_img = cv.cvtColor(cropped_img, cv.COLOR_BGR2RGB)
+        # do augmentations with keypoints
+        if self.transform:
+            transformed = self.transform(image=cropped_img, keypoints=cropped_keypoints)
+            assert np.array(transformed['keypoints']).shape == np.random.rand(9,2).shape
```

Comment:
shape is just a tuple, so I guess we don't need to generate a random array every time